File: OpenHarmony/developtools/hapsigner/hapsigntool_cpp/hap/entity/src/sign_block_data.cpp
/*
 * Copyright (c) 2024-2024 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

#include "sign_block_data.h"

namespace OHOS {
namespace SignatureTools {

SignBlockData::SignBlockData(const std::vector<int8_t>& signData, const char type)
{
    m_signData = std::move(signData);
    m_type = type;
    m_len = signData.size();
    m_isByte = true;
}

SignBlockData::SignBlockData(const std::string& signFile, const char type)
{
    m_signFile = std::move(signFile);
    m_type = type;
    m_len = FileUtils::GetFileLen(signFile);
    m_isByte = false;
}

char SignBlockData::GetType()
{
    return m_type;
}

std::vector<int8_t>& SignBlockData::GetBlockHead()
{
    return m_blockHead;
}

void SignBlockData::SetBlockHead(const std::vector<int8_t>& blockHead)
{
    m_blockHead = std::move(blockHead);
}

std::vector<int8_t>& SignBlockData::GetSignData()
{
    return m_signData;
}

std::string SignBlockData::GetSignFile()
{
    return m_signFile;
}

long SignBlockData::GetLen()
{
    return m_len;
}

bool SignBlockData::GetByte()
{
    return m_isByte;
}

} // namespace SignatureTools
} // namespace OHOS

File: OpenHarmony/developtools/hapsigner/hapsigntool/hap_sign_tool_lib/src/main/java/com/ohos/hapsigntool/hap/entity/SignBlockData.java
/*
 * Copyright (c) 2023-2023 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.ohos.hapsigntool.hap.entity;

import com.ohos.hapsigntool.utils.FileUtils;

/**
 * sign block data
 *
 * @since 2023-11-07
 */
public class SignBlockData {
    private char type;
    private byte[] blockHead;
    private byte[] signData;
    private String signFile;
    private long len;
    private boolean isByte;

    public SignBlockData(byte[] signData, char type) {
        this.signData = signData;
        this.type = type;
        this.len = signData == null ? 0 : signData.length;
        this.isByte = true;
    }

    public SignBlockData(String signFile, char type) {
        this.signFile = signFile;
        this.type = type;
        this.len = FileUtils.getFileLen(signFile);
        this.isByte = false;
    }

    public char getType() {
        return type;
    }

    public void setType(char type) {
        this.type = type;
    }

    public byte[] getBlockHead() {
        return blockHead;
    }

    public void setBlockHead(byte[] blockHead) {
        this.blockHead = blockHead;
    }

    public byte[] getSignData() {
        return signData;
    }

    public void setSignData(byte[] signData) {
        this.signData = signData;
    }

    public String getSignFile() {
        return signFile;
    }

    public void setSignFile(String signFile) {
        this.signFile = signFile;
    }

    public long getLen() {
        return len;
    }

    public void setLen(long len) {
        this.len = len;
    }

    public boolean isByte() {
        return isByte;
    }

    public void setByte(boolean isByte) {
        this.isByte = isByte;
    }
}
